### PR TITLE
fix: tolerate strange cases in container addressing

### PR DESCRIPTION
**Pull request by stoksc**
## Description
A lot time ago, Docker started returning 2 port bindings for each port, one for ipv4 and one for ipv6. This caused the rendezvous code to explode because suddenly it was getting 2 addresses for each binding where it expected one. To fix this, we preferred the binding that matched the what the agent was using to talk to us.

Three years later, @wes-turner's dev machine has the agent connecting through `::1` and Docker only publishing ipv4 port bindings, and so his experiment fail to find any rendezvous addresses. This reverts back to the old logic then adds a dedup instead, so wes's machine should find some address either way (fyi this doesn't really matter at all, because the trial doesn't use these rendezvous addresses anyway since it is a single trial experiment, it just explodes without them anyway.. but at least now some folks dev machines will work). 

## Test Plan
- [x] has tests, dont manually test

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
RM-52

### CODE REVIEW

File: master/pkg/aproto/master_message.go [hunk 1]
```diff
@@ -2,11 +2,10 @@ package aproto
 
 import (
 	"fmt"
-	"net"
 	"strconv"
 	"time"
 
-	"github.com/pkg/errors"
+	"golang.org/x/exp/maps"
```
Comment (NicholasBlaskey) [inline master/pkg/aproto/master_message.go:8]: create a ticket for banning ```"golang.org/x/exp/maps"``` in favor of ```maps```?

also slices?

I don't remember what we said when we talked about this before

File: master/pkg/aproto/master_message.go [hunk 2]
```diff
@@ -113,47 +110,39 @@ func (c ContainerStarted) Addresses() []cproto.Address {
...
-		ipAddresses := make([]string, 0, len(networks))
-		for _, network := range networks {
-			ipAddresses = append(ipAddresses, network.IPAddress)
+
+		// We used to return a cproto.Address for each network in info.NetworkSettings.Networks that existed, but the
+		// harness only wants the tuple (host_ip, host_port) and disregards container_ip, and (!!) if it does receive
+		// two cproto.Address mappings for the rendezvous port of a container, it would just explode anyway because it
+		// expects a single address per container and doesn't know which one to use.
+		containerIPOnNetwork := "missing"
+		if containerNetwork := info.NetworkSettings.Networks[networkMode.NetworkName()]; containerNetwork != nil {
```
Comment (NicholasBlaskey) [inline master/pkg/aproto/master_message.go:119]: to be honest I can't verify this is correct, I'm not saying this is wrong

I'm just not sure if ```info.NetworkSettings.Networks``` always will exist?

and the harness is okay with getting only one address in some cases?

Comment (NicholasBlaskey) [inline master/pkg/aproto/master_message.go:119]: also can we delete containerIP eventually? I know you were implying it was unused, I haven't looked at usages

**Review by NicholasBlaskey — APPROVED** [review]

File: master/pkg/aproto/master_message.go (hunk 1 above)
Comment (stoksc) [reply master/pkg/aproto/master_message.go:8]: maps doesn't have eveything x/exp/maps has yet.

File: master/pkg/aproto/master_message.go (hunk 2 above)
Comment (stoksc) [reply master/pkg/aproto/master_message.go:119]: yeah it's unused.

Comment (stoksc) [reply master/pkg/aproto/master_message.go:119]: nothing ever uses containerIP anyway. i don't suspect it will.

**Comment by stoksc** [comment]
i've decided we shouldn't land this after thinking on it

**Comment by stoksc** [comment]
@wes-turner maybe we can chat later, i'll respond on the ticket though with a different fix.

**Pull request closed without merging**